storyblok/field-plugin
# feat(common): pass manifest json to sandbox

## What?
It allows the Sandbox/Container to bootstrap options from a manifest file located within a field-plugin root folder.

## Why?
JIRA: EXT-1877

[see the RFC](https://github.com/storyblok/field-plugin/blob/main/rfcs/000-persist-options-and-values.md)

When our Vite `printDev` plugin finds the manifest file:
![Screenshot 2023-09-18 at 08 49 36](https://github.com/storyblok/field-plugin/assets/1240591/58c9e676-fcde-4918-be02-746860c29d7f)

When our Vite `printDev` plugin doesn't find the manifest file:
![Screenshot 2023-09-18 at 08 52 17](https://github.com/storyblok/field-plugin/assets/1240591/423c4cda-d243-4f5e-9b10-f5e26420e280)

When our Vite `printDev` plugin finds the manifest file but it's not valid:
![Screenshot 2023-09-18 at 08 58 45](https://github.com/storyblok/field-plugin/assets/1240591/45472416-f574-4743-b66f-4a883aabdb75)

## How to test? (optional)

From the root folder of this repository, run:

1. `yarn dev:container`
2. `yarn dev:react`
3. Open [Sandbox](http://localhost:7070/field-plugin?url=http%3A%2F%2Flocalhost%3A8080%2F&manifest=%7B%22options%22%3A%5B%7B%22name%22%3A%22option1%22%2C%22value%22%3A%22value1%22%7D%2C%7B%22name%22%3A%22option2%22%2C%22value%22%3A%22value2%22%7D%5D%7D)

It should load the options automatically (from the manifest file) like below:
![image](https://github.com/storyblok/field-plugin/assets/1240591/686d865b-1b53-41e2-911b-f28733d23857)

If you want to change the manifest options locally, edit this file here [React Manifest](https://github.com/storyblok/field-plugin/blob/EXT-1877-pass-manifest-json-to-sandbox/packages/cli/templates/react/field-plugin.config.json)
